### PR TITLE
[New] Add Type definitions to minecraft-pinger

### PR DESCRIPTION
**Pull request by Grayson-code**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by Grayson-code** [comment]
i am posting this 1 hour before Exam so extra credit for that IG

File: types/minecraft-pinger/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
+// Type definitions for minecraft-pinger 1.0
+// Project: https://github.com/dennisbruner/node-minecraft-pinger
+// Definitions by: Grayson-code <https://github.com/Grayson-code>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export interface pingPromiseInterface {
```
Comment (peterblazejewicz) [inline types/minecraft-pinger/index.d.ts:6]: this is called either `data` or `result` directly in the source code:
this is called `result` in the readme (public api):
https://github.com/dennisbruner/node-minecraft-pinger#example
Please use CamelCase notation for interfaces, most common notation pattern in JS:
https://developer.mozilla.org/en-US/docs/Web/API#interfaces

File: types/minecraft-pinger/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+     *  The Port of the server , most servers default to 25565
+     */
+    port: number
+  ): pingPromiseInterface;
+  /**
+   * Checks The Service Record
+   *
+   * The URL or hostname is the hostname , eg: https://example.com
+   */
+  export function checkSrvRecord(hostname: string): Promise<iCheckSrvRecord>;
```
Comment (peterblazejewicz) [inline types/minecraft-pinger/index.d.ts:133]: the module does not export 'checkSrvRecord', so this looks incorrect, only `ping`, `pingPromise`

File: types/minecraft-pinger/minecraft-pinger-tests.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// Copyright (c) 2022 Northern Star
+//
+// This software is released under the MIT License.
+// https://opensource.org/licenses/MIT
+
```
Comment (peterblazejewicz) [inline types/minecraft-pinger/minecraft-pinger-tests.ts:5]:
```suggestion
```
optionally, unless you're under some obligation to have this copy left banner here

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]
Per peter's comments

File: types/minecraft-pinger/minecraft-pinger-tests.ts (hunk 3 above)
Comment (Grayson-code) [reply types/minecraft-pinger/minecraft-pinger-tests.ts:5]: Oh yeah Forgot that was there

File: types/minecraft-pinger/index.d.ts (hunk 2 above)
Comment (Grayson-code) [reply types/minecraft-pinger/index.d.ts:133]: Yeah was kinda confused on that

File: types/minecraft-pinger/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+     * The Host Name
+     */
+    hostname: string,
+    /**
+     *  The Port of the server , most servers default to 25565
+     */
+    port: number
+  ): Data;
+
+export as namespace minecraftpinger;
```
Comment (peterblazejewicz) [inline types/minecraft-pinger/index.d.ts:119]:
```suggestion
```
this seems to be pure, Node only package, that export make it compatible with global script usage, which can lead to wrong usage.
Can you now also run (from your repo root directory):
```bash
npm run prettier -- --write types/minecraft-pinger
```
(or just format white space with your IDE features)
thx!

Comment (Grayson-code) [reply types/minecraft-pinger/index.d.ts:119]: Yeah Sure!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@Grayson-code thx!

**Review by RyanCavanaugh — APPROVED** [review]

**Comment by Grayson-code** [comment]
Ready to merge

**Pull request merged**